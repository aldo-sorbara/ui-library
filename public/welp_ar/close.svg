<svg xmlns="http://www.w3.org/2000/svg"
    fill="currentColor"
    height="24"
    width="24"
    viewBox="0 0 35 35">
  <path d="M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z"/>
</svg>
