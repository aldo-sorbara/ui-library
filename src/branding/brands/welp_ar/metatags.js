export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
