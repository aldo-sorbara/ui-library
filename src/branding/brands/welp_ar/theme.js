import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
