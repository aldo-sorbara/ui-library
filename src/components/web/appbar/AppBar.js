import React from 'react';
import PropTypes from 'prop-types';
import { withTheme } from 'styled-components';

import styled from 'styled-components';

const AppBar = props => {
  const imgURI = props.inverse
    ? props.theme.assetsPaths.logoInverse
    : props.theme.assetsPaths.logo;
  return (
    <StyledAppBar {...props}>
      <div>
        <img src={imgURI} alt={BRAND} height='32' />
      </div>
    </StyledAppBar>
  );
};

/* istanbul ignore next */
const BRAND = process.env.REACT_APP_BRAND
  ? process.env.REACT_APP_BRAND
  : 'welp_ar';

AppBar.defaultProps = {
  inverse: false,
  sticky: false
};

AppBar.propTypes = {
  inverse: PropTypes.bool.isRequired,
  sticky: PropTypes.bool.isRequired
};

const StickyAppBar = () => `
  width: 100%;
  top:0;
  left:0;
  color: white;
`;

const StyledAppBar = styled.div`
  font-size: 0;
  ${props => props.sticky && StickyAppBar};
  padding: ${props => props.theme.spacer.header.inner};
  position: ${props => (props.sticky ? 'fixed' : 'relative')};
  z-index: ${props => props.theme.zIndex.appBar};

  > img {
    margin-top: ${props => props.theme.spacer.medium};
  }
`;

StyledAppBar.defaultProps = {
  theme: {
    assetsPaths: {
      logoInverse: './welp_ar/logo-inverse.png',
      logo: './welp_ar/logo.png'
    },
    zIndex: {
      appBar: 120
    },
    spacer: {
      medium: '1rem',
      header: {
        inner: '.5rem 1rem'
      }
    }
  }
};

export default withTheme(AppBar);
