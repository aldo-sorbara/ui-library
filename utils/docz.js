export const styles = {
  display: 'flex',
  flex: 1,
  flexDirection: 'row',
  flexWrap: 'wrap',
  justifyContent: 'space-around'
};
