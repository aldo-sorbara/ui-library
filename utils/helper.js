import { shallow, mount } from 'enzyme';

import { ThemeConsumer } from 'styled-components';

export const shallowWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return shallow(children);
};

export const mountWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return mount(children);
};
