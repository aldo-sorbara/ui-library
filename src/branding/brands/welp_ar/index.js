import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
