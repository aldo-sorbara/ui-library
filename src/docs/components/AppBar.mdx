---
name: AppBar
menu: Components
route: /appbar
---

import { Playground, PropsTable } from 'docz';
import { Fragment } from 'react';
import AppBar from '../../components/web/appbar';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../branding/brands/welp_ar';
import GlobalStyle from '../../components/web/globalstyle';
import { styles } from '../../../utils/docz';

# AppBar

<PropsTable of={AppBar} />

## Basic usage

<Playground>
  <ThemeProvider theme={welp_ar.theme}>
    <>
      <GlobalStyle theme={welp_ar.theme} />
      <AppBar />
      <AppBar inverse style={{ background: '#000000' }} />
      <AppBar sticky style={{ background: '#FFFFFF', left: 'auto' }} />
    </>
  </ThemeProvider>
</Playground>

[Try it on Storybook!](./storybook/index.html?path=/story/appbar--with-default-props)
