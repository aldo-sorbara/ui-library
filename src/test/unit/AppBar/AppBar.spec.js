import React from 'react';
import AppBar from '../../../components/web/appbar';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

describe('AppBar component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <AppBar className='test-appBar' />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render a default AppBar', () => {
    const wrapper = mountWithTheme(
      <AppBar inverse={false} sticky={false} className='test-appbar' />,
      welp_ar.theme
    );
    expect(wrapper.props().inverse).toBe(false);
    expect(wrapper.props().sticky).toBe(false);
  });
  it('should render an inverse AppBar', () => {
    const wrapper = mountWithTheme(
      <AppBar inverse className='inverse-appBar' />,
      welp_ar.theme
    );
    expect(wrapper.props().inverse).toBe(true);
  });
  it('should render a sticky AppBar', () => {
    const wrapper = mountWithTheme(
      <AppBar sticky className='sticky-appBar' />,
      welp_ar.theme
    );
    expect(wrapper.props().sticky).toBe(true);
    expect(wrapper).toHaveStyleRule('width', '100%');
    expect(wrapper).toHaveStyleRule('top', '0');
    expect(wrapper).toHaveStyleRule('left', '0');
  });
});
