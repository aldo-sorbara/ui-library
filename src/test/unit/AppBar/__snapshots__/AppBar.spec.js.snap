// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar component should render snapshot 1`] = `
.c0 {
  font-size: 0;
  padding: .5rem 1rem;
  position: relative;
  z-index: 120;
}

.c0 > img {
  margin-top: 1rem;
}

<div
  className="test-appBar c0"
>
  <div>
    <img
      alt="welp_ar"
      height="32"
      src="./public/welp_ar/logo.png"
    />
  </div>
</div>
`;
